File: assets/process_images.js
const fs = require('fs');
const path = require('path');
const sharp = require('sharp');

const artifactDir = "C:\\Users\\admin\\.gemini\\antigravity\\brain\\4b91b6e1-a5f4-43f4-8434-45e18ef709e8";
const outputDir = "C:\\Users\\admin\\Desktop\\Detective agency\\assets\\images";

if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

// Find all generated images
const files = fs.readdirSync(artifactDir);
const imageFiles = files.filter(f => f.endsWith('.jpg') || f.endsWith('.png') || f.endsWith('.webp'));

async function processImage(filename) {
    const imgPath = path.join(artifactDir, filename);
    let baseName = filename.substring(0, filename.lastIndexOf('.'));
    if (baseName.includes('_17')) {
        baseName = baseName.substring(0, baseName.lastIndexOf('_'));
    }
    const outputPath = path.join(outputDir, `${baseName}.webp`);

    try {
        let quality = 80;
        let info;
        
        // Resize first to ensure it's not massive
        let buffer = await sharp(imgPath)
            .resize({ width: 1200, withoutEnlargement: true })
            .webp({ quality: quality })
            .toBuffer({ resolveWithObject: true });
            
        // Binary search for quality to keep under 95KB
        if (buffer.info.size > 95 * 1024) {
            let low = 10, high = 80;
            let bestBuffer = buffer.data;
            while (low <= high) {
                let mid = Math.floor((low + high) / 2);
                let tempBuffer = await sharp(imgPath)
                    .resize({ width: 1200, withoutEnlargement: true })
                    .webp({ quality: mid })
                    .toBuffer({ resolveWithObject: true });
                
                if (tempBuffer.info.size > 95 * 1024) {
                    high = mid - 1;
                } else {
                    bestBuffer = tempBuffer.data;
                    low = mid + 1;
                }
            }
            fs.writeFileSync(outputPath, bestBuffer);
            console.log(`Processed: ${baseName}.webp - Size: ${(bestBuffer.length / 1024).toFixed(2)} KB`);
        } else {
            fs.writeFileSync(outputPath, buffer.data);
            console.log(`Processed: ${baseName}.webp - Size: ${(buffer.info.size / 1024).toFixed(2)} KB`);
        }
        
    } catch (e) {
        console.error(`Failed to process ${filename}:`, e);
    }
}

async function run() {
    for (const file of imageFiles) {
        await processImage(file);
    }
    console.log("Done processing images.");
}

run();

File: assets/process_images.py
import os
import glob
from PIL import Image

artifact_dir = r"C:\Users\admin\.gemini\antigravity\brain\4b91b6e1-a5f4-43f4-8434-45e18ef709e8"
output_dir = r"C:\Users\admin\Desktop\Detective agency\assets\images"

os.makedirs(output_dir, exist_ok=True)

# Find all generated images
image_files = glob.glob(os.path.join(artifact_dir, "*.jpg"))
image_files.extend(glob.glob(os.path.join(artifact_dir, "*.png")))

def process_image(img_path):
    filename = os.path.basename(img_path)
    # Remove the timestamp suffix like _1785577721698
    base_name = filename.rsplit('_', 1)[0] if '_' in filename else os.path.splitext(filename)[0]
    output_path = os.path.join(output_dir, f"{base_name}.webp")
    
    try:
        with Image.open(img_path) as img:
            # Resize if too large to help compression
            img.thumbnail((1200, 1200), Image.Resampling.LANCZOS)
            
            # Binary search for quality to keep under 100KB
            low, high = 10, 95
            best_quality = 50
            
            while low <= high:
                mid = (low + high) // 2
                img.save(output_path, "WEBP", quality=mid)
                size_kb = os.path.getsize(output_path) / 1024
                
                if size_kb > 95:
                    high = mid - 1
                else:
                    best_quality = mid
                    low = mid + 1
                    
            # Final save with best found quality
            img.save(output_path, "WEBP", quality=best_quality)
            print(f"Processed: {base_name}.webp - Size: {os.path.getsize(output_path) / 1024:.2f} KB")
    except Exception as e:
        print(f"Failed to process {filename}: {e}")

for img in image_files:
    process_image(img)
